lib/diagnostic.js

```diff
@@ -5,17 +5,31 @@
 // the minimum value in an array
 // this callback will be passed by the tests to the reduce method, e.g.
 // `let minimum = array.reduce(min);`
-const min = function (/* previous, current */) {
-
+const min = function (previous, current) {
+  let result = 0;
+  for (let i = 0; i < previous.length; i++) {
+    if (previous[i] > result) {
+      result = previous[i];
+    }
+  }
+  return result;
 };
 
 // Question 2
 // Write a callback for reduce that produces
 // the sum of the values in the array
 // this callback will be passed by the tests to the reduce method, e.g.
 // `let total = array.reduce(sum);`
-const sum = function (/* previous, current */) {
-
+const sum = function (previous, current, callback) {
```

Comment:
```callback``` doesn't need to be a parameter since the ```sum()``` itself is being used as a callback for ```reduce```.  In fact, in that particular context, ```callback``` will be treated as an array since ```reduce```'s callback parameters are the current element being iterated over, the index of the element, and the current working array.